packages/p2p/src/components/buy-sell/buy-sell-form.jsx

```diff
@@ -112,6 +102,18 @@ const BuySellForm = props => {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [input_amount, effective_rate]);
 
+    React.useEffect(() => {
+        if (did_mount_ref.current) {
+            showModal({
+                key: 'RateChangeModal',
+                props: {
+                    currency: buy_sell_store.local_currency,
+                },
+            });
+        }
+        did_mount_ref.current = true;
+    }, [exchange_rate, did_mount_ref]);
```

Comment:
Please recheck what @farzin-deriv commented. Ideally currency from exchange rate should be in this dependency. You should store a state on the previous currency in this hook. If it is different from the one from the exchange rates hook, show the modal.